GradeCenter.Services/Schools/SchoolService.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using GradeCenter.Data.Models;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using GradeCenter.Data;
+
+namespace GradeCenter.Services.Schools
+{
+    public class SchoolService : ISchoolService
+    {
+        private readonly GradeCenterContext _db;
+        public SchoolService(GradeCenterContext gradeCenterContext)
+        {
+            _db = gradeCenterContext;
+        }
+
+        /// <summary>
+        /// Takes an id as a paramater
+        /// and returns the associated 
+        /// School entity instance.
+        /// </summary>
+        /// <param name="id"></param>        
+        /// <returns></returns>
+        public School? GetSchoolById(int id)
+        {
+            var school = _db?.Schools?.FirstOrDefault(school => school.Id == id);
+
+            return school;
+        }
+
+        /// <summary>
+        /// Creates a new School entity instance
+        /// in the database. Whilst ensuring there is a 
+        /// relationship with a principal entity.
+        /// </summary>
+        /// <param name="newName"></param>
+        /// <param name="newAddress"></param>
+        /// <param name="principalId"></param>
+        /// <param name="principalFirstName"></param>
+        /// <param name="principalLastName"></param>
+        /// <returns></returns> 
+        public async Task Create(string newName, string newAddress, int principalId, string principalFirstName, string principalLastName)
+        {
+            var principal = _db?.Principals?.FirstOrDefault(principal => principal.Id == principalId);
+            principal ??= EnsurePrincipalCreated(principalFirstName, principalLastName).GetAwaiter().GetResult();
```

Comment:
- You should use "await" here
- I don't think you need to ensure the principle is created here, so you can remove this line